### DISCUSSION TITLE
[Question] Terrain with multiple physics and visual materials

### DISCUSSION THREAD

**Discussion by titoirfan** [isaac-sim/IsaacLab#1160] [Q&A]
### Question

Hi, thanks for the great work on Orbit.

I am planning to do vision-based locomotion on Orbit. 

To that end, I need a terrain that has different physics and visual materials on each sub-terrain (or on different parts), so that the robot can visually distinguish different materials and learn to adapt to the associated physical properties.

However, If I understand correctly, we can not have multiple materials within a single `TerrainImporterCfg` (I am currently using it alongside `InteractiveScene` to build the terrain).

Do you have any suggestion on how I can do this? 

---

I am currently thinking of using multiple adjacent `TerrainImporterCfg` (each with different material). 
However, I am not sure if this is doable, as I am unsure if the origin point of each terrain in the scene is configurable.

What do you think about this? Do you think there is a better way to do it?

Thanks for your time and support.

**Comment by Dhoeller19** [comment]
Hi @titoirfan,
Indeed, you will have to make your own TerrainImporter and TerrainGenerator.
The terrain generator should return individual meshes, and in the terrain importer you should convert each individual mesh to a prim with its own physics material.
